File: src/main/dialogs/LegendDialog.java
package main.dialogs;

import javax.swing.*;
import java.awt.*;
import java.util.Map;

public class LegendDialog extends JDialog {

    public LegendDialog(Window parent, String title, Map<String, Color> legend) {
        super(parent, title);
        this.setTitle(title);
        getContentPane().setBackground(Color.WHITE);
        setLayout(new GridLayout(legend.size(), 1, 2, 2));
        setDefaultCloseOperation(DISPOSE_ON_CLOSE); // Note: JDialog doesn't have EXIT_ON_CLOSE usually, but DISPOSE is
                                                    // better. KT used DISPOSE.

        for (Map.Entry<String, Color> entry : legend.entrySet()) {
            add(createDialogBox(entry.getKey(), entry.getValue()));
        }
        pack();
        setResizable(false);
        setAlwaysOnTop(true);
        setLocationRelativeTo(parent);
    }

    private JPanel createDialogBox(String string, Color color) {
        JPanel jPanel = new JPanel(new FlowLayout(FlowLayout.LEFT));
        JLabel jLabel = new JLabel(string);
        // jLabel.setBackground(Color.BLACK); // JLabel background only works if opaque
        // jLabel.setOpaque(true);
        // usage in KT: jLabel.background= Color.BLACK

        jLabel.setFont(new Font(Font.SANS_SERIF, Font.BOLD, 20));
        jLabel.setForeground(color);

        jPanel.setBackground(Color.BLACK);
        jPanel.add(jLabel);
        return jPanel;
    }
}

File: src/pages/dialogs/LegendDialog.kt
package pages.dialogs

import java.awt.Color
import java.awt.FlowLayout
import java.awt.Font
import java.awt.GridLayout
import java.awt.Window
import javax.swing.JDialog
import javax.swing.JLabel
import javax.swing.JPanel

class LegendDialog(parent: Window?, title: String?, legend:Map<String, Color>): JDialog(parent,title){
    init {
        this.title=title
        contentPane.background= Color.WHITE
        layout= GridLayout(legend.size, 1, 2, 2)
        defaultCloseOperation=DISPOSE_ON_CLOSE
        for(entry in legend.entries){
            add(createDialogBox(entry.key,entry.value))
        }
        pack()
        isResizable=false
        isAlwaysOnTop=true
        setLocationRelativeTo(parent)
    }
    private fun createDialogBox(string: String,color: Color): JPanel {
        val jPanel= JPanel(FlowLayout(FlowLayout.LEFT))
        val jLabel= JLabel(string)
        jLabel.background= Color.BLACK
        jLabel.font= Font(Font.SANS_SERIF, Font.BOLD, 20)
        jLabel.foreground= color
        jPanel.background= Color.BLACK
        jPanel.add(jLabel)
        return jPanel
    }
}